openedx/tests/xblock_integration/test_done.py

```diff
@@ -1,269 +1,67 @@
-        {'email': 'view@test.com', 'password': 'foo'},
-    ]
-
-    @classmethod
-    def setUpClass(cls):
-        """
-        Create a page with two of the XBlock on it
-        """
-        # Nose runs setUpClass methods even if a class decorator says to skip
-        # the class: https://github.com/nose-devs/nose/issues/946
-        # So, skip the test class here if we are not in the LMS.
-        if settings.ROOT_URLCONF != 'lms.urls':
-            raise unittest.SkipTest('Test only valid in lms')
-
-        super(TestDone, cls).setUpClass()
-        cls.course = CourseFactory.create(
-            display_name='Done_Test_Course'
-        )
-        with cls.store.bulk_operations(cls.course.id, emit_signals=False):
-            cls.chapter = ItemFactory.create(
-                parent=cls.course,
-                display_name='Overview',
-                category='chapter'
-            )
-            cls.section = ItemFactory.create(
-                parent=cls.chapter,
-                display_name='Welcome',
-                category='sequential'
-            )
-            cls.unit = ItemFactory.create(
-                parent=cls.section,
-                display_name='New Unit',
-                category='vertical'
-            )
-            cls.xblock1 = ItemFactory.create(
-                parent=cls.unit,
-                category='done',
-                display_name='done_0'
-            )
-            cls.xblock2 = ItemFactory.create(
-                parent=cls.unit,
-                category='done',
-                display_name='done_1'
-            )
-
-        cls.course_url = reverse(
-            'courseware_section',
-            kwargs={
-                'course_id': unicode(cls.course.id),
-                'chapter': 'Overview',
-                'section': 'Welcome',
-            }
-        )
-
-    def setUp(self):
-        """
-        Create users
-        """
-        super(TestDone, self).setUp()
-        for idx, student in enumerate(self.STUDENTS):
-            username = "u{}".format(idx)
-            self.create_account(username, student['email'], student['password'])
-            self.activate_user(student['email'])
 
-        self.staff_user = GlobalStaffFactory()
+    olx_scenarios = {  # Currently not used
+        "two_done_block_test_case": """<vertical>
+          <done urlname="done0"/>
+          <done urlname="done1"/>
+        </vertical>"""
+    }
+
+    # This is a stop-gap until we can load OLX and/or OLX from
+    # normal workbench scenarios
+    test_configuration = [
+        {
+            "urlname": "two_done_block_test_case",
+            "olx": olx_scenarios[0],
+            "xblocks": [  # Stopgap until we handle OLX
+                {
+                    'blocktype': 'done',
+                    'urlname': 'done_0'
+                },
+                {
+                    'blocktype': 'done',
+                    'urlname': 'done_1'
+                }
+            ]
+        }
```

Comment:
If there is a sensible way to do that, I would be very happy. It would simplify XBlock authoring and testing significantly. I'd need significant help; I don't understand any of that code at all.